### ISSUE TITLE
Errors not reported when signature help is active

### ISSUE THREAD

**Issue by DanielRosenwasser** [microsoft/TypeScript#52369]
```ts
declare function printValue(str: string): void;
declare function printValue(num: number, maxFractionDigits?: number): void

printValue("hello"/*$*/)
```

At `/*$*/`, type in `, 123`. Notice that you're not getting any errors.

Now try **dismissing signature help** and making a small edit like erasing `3` or inserting `4`. An error will appear.

[signatureHelpPoisoningTypeCheckerCache.webm](https://user-images.githubusercontent.com/972891/214007263-388e148c-9e02-4181-96e1-b73930620cdd.webm)

What are the mechanics here that're making this happen? If you're using VS Code, that initial `,` will start a signature help session. Every subsequent character will likey be used as a retrigger for signature help prior to requesting diagnostics. And so signature help, which I recall is checked in a specific way to handle overloads that differ slightly in arity, keeps getting requested for a specific expression before the rest of the type-checker gets to walk over it to report errors.

I believe that because signature help gets checked in a special way, it is likely poisoning the type-checker state we use for reporting errors as well.

**Comment by andrewbranch** [comment]
Duplicate of https://github.com/microsoft/TypeScript/issues/49624
